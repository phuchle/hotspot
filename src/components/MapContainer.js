import React, { Component } from 'react';
import { withGoogleMap, GoogleMap, Circle } from 'react-google-maps';
import { Grid, Row, Col } from 'react-bootstrap';
import Sidebar from './Sidebar';
import PropTypes from 'prop-types';

const MapWrapper = withGoogleMap(props => {
  return (
    <GoogleMap
      defaultZoom={14}
      center={props.defaultCenter}
    >
      {props.location.state.hotspots.map(hotspot => {
        let options;
        if (hotspot.venue.id === props.activeHotspot) {
          options = {
            fillColor: 'red',
            fillOpacity: .8,
            strokeColor: 'red',
            strokeWeight: 2
          };
        } else {
          options = {
            fillColor: 'blue',
            fillOpacity: .3,
            strokeColor: '#333',
            strokeWeight: .5
          };
        }

        return (
          <Circle
            key={hotspot.venue.id}
            center={ {lat: hotspot.venue.location.lat, lng: hotspot.venue.location.lng} }
            radius={hotspot.venue.stats.checkinsCount / 100}
            options={options}
          />
        );
      })}
    </GoogleMap>
  );
});

class MapContainer extends Component {
  constructor(props) {
    super(props);

    this.state = {
      activeHotspot: null,
      mapCenter: this.props.location.state.mapCenter,
    };

    this.setHotspot = this.setHotspot.bind(this);
  }
  setHotspot(venue) {
    this.setState({
      activeHotspot: venue.id,
      mapCenter: { lat: venue.location.lat, lng: venue.location.lng }
    });
  }
  render() {
    return (
      <Grid>
        <Row>
          <Col xs={12} sm={8}>
            <MapWrapper
              activeHotspot={this.state.activeHotspot}
              defaultCenter={this.state.mapCenter}
              location={this.props.location}
              containerElement={<div className="map-container" />}
              mapElement={<div className="map-container" />}
            />
          </Col>
          <Col xs={12} sm={4}>
            <Sidebar
              setHotspot={this.setHotspot}
              hotspots={this.props.location.state.hotspots}
              destination={this.props.location.state.destination}
              location={this.props.location.state.location}
              style={{
                overflowY: 'scroll'
              }}
            />
          </Col>
        </Row>
      </Grid>
    );
  }
};

MapContainer.propTypes = {
  location: PropTypes.object.isRequired,
};

export default MapContainer;
