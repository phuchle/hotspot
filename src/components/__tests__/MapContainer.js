import React from 'react';
import renderer from 'react-test-renderer';
import MapContainer from '../MapContainer';

describe('MapContainer', () => {
  test('renders without crashing', () => {
    const component = renderer.create(
      <MapContainer />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
