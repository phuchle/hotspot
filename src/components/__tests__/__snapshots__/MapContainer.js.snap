// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapContainer renders without crashing 1`] = `
<div
  className="map-container"
>
  <div
    className="map-container"
  />
  <div />
</div>
`;
